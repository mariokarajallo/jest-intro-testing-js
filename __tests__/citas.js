import Citas from "../js/Clases/Citas";

//Instanciamos por fuera de los test para poder utilizar en varias pruebas
const citas = new Citas();
const id = Date.now();
describe("Probar la clase de citas", () => {
  test("Agregar una nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Frodo",
      propietario: "Mario",
      telefono: "111222",
      fecha: "11-06-23",
      hora: "10:40",
      sintomas: "Duerme mucho",
    };

    citas.agregarCita(citaObj);

    //prueba
    expect(citas).toMatchSnapshot();
  });

  test("Actualizar cita", () => {
    const citaActualizada = {
      id,
      mascota: "Kira",
      propietario: "Mario",
      telefono: "111222",
      fecha: "11-06-23",
      hora: "10:40",
      sintomas: "Duerme mucho",
    };

    citas.editarCita(citaActualizada);
    expect(citas).toMatchSnapshot();
  });
});
