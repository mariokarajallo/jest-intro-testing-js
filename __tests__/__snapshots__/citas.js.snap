// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "11-06-23",
      "hora": "10:40",
      "id": 1686540398451,
      "mascota": "Kira",
      "propietario": "Mario",
      "sintomas": "Duerme mucho",
      "telefono": "111222",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "11-06-23",
      "hora": "10:40",
      "id": 1686540398451,
      "mascota": "Frodo",
      "propietario": "Mario",
      "sintomas": "Duerme mucho",
      "telefono": "111222",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliiminar cita 1`] = `
Citas {
  "citas": [],
}
`;
